### PR TITLE
rbd: add `mkfsOptions` to the StorageClass and pass them to `mkfs`

### PR DESCRIPTION
**Pull request by nixpanic**
Add `mkfsOptions` to the StorageClass and pass them to the `mkfs`
command while creating the filesystem on the RBD device.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: internal/rbd/nodeserver.go [hunk 1]
```diff
@@ -771,34 +780,43 @@ func (ns *NodeServer) mountVolumeToStagePath(
...
+		mkfs := "mkfs." + fsType
 		switch fsType {
 		case "ext4":
-			args = []string{"-m0", "-Enodiscard,lazy_itable_init=1,lazy_journal_init=1"}
 			if fileEncryption {
 				args = append(args, "-Oencrypt")
 			}
-			args = append(args, devicePath)
 		case "xfs":
-			args = []string{"-K", devicePath}
```
Comment (Rakshith-R) [inline internal/rbd/nodeserver.go:788]: "-K" also needs to move to  newly added mkfsDefultArgs ? 
```
mkfsDefaultArgs = map[string][]string{
		"ext4": {"-m0", "-Enodiscard,lazy_itable_init=1,lazy_journal_init=1"},
		"xfs": {"-K"},
	}
```

Comment (nixpanic) [reply internal/rbd/nodeserver.go:788]: yes, good catch!

**Review by Rakshith-R — APPROVED** [review]

File: e2e/rbd.go [hunk 2]
```diff
@@ -1089,6 +1089,45 @@ var _ = Describe("RBD", func() {
 				}
 			})
 
+			By("create a PVC and bind it to an app with ext4 as the FS and 1024 inodes ", func() {
```
Comment (humblec) [inline e2e/rbd.go:1092]: @nixpanic can we have  a test case for XFS FS as well ?

File: internal/rbd/nodeserver.go [hunk 3]
```diff
@@ -101,6 +101,15 @@ var (
 	// xfsHasReflink is set by xfsSupportsReflink(), use the function when
 	// checking the support for reflink.
 	xfsHasReflink = xfsReflinkUnset
+
+	mkfsDefaultArgs = map[string][]string{
+		"ext4": {"-m0", "-Enodiscard,lazy_itable_init=1,lazy_journal_init=1"},
```
Comment (humblec) [inline internal/rbd/nodeserver.go:106]: having some references to these options and why we enable them as default options could help

Comment (nixpanic) [reply internal/rbd/nodeserver.go:106]: sure, but I don't know where they come from. This PR just moves them around...

**Review by yati1998 — APPROVED** [review]

File: e2e/rbd.go (hunk 2 above)
Comment (nixpanic) [reply e2e/rbd.go:1092]: I do not think there is much value in that. XFS is used in other tests already, so this PR does not break it.

**Review by humblec — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Pull request merged**